x-pack/filebeat/input/azureeventhub/config_test.go

```diff
@@ -55,3 +55,92 @@ func TestValidate(t *testing.T) {
+		config := defaultConfig()
+		config.ProcessorVersion = "v1"
+		config.ConnectionString = "Endpoint=sb://my-namespace.servicebus.windows.net/;SharedAccessKeyName=my-key;SharedAccessKey=my-secret"
+		config.EventHubName = "my-event-hub"
+		config.SAName = "teststorageaccount"
+		config.SAKey = "my-secret"
+		config.SAContainer = "filebeat-activitylogs-event_hub_00"
+
+		if err := config.Validate(); err != nil {
+			t.Fatalf("unexpected validation error: %v", err)
+		}
+
+		assert.Nil(t, config.ConnectionStringProperties.EntityPath)
+	})
+}
+
+func TestValidateConnectionStringV2(t *testing.T) {
+	t.Run("Connection string contains entity path", func(t *testing.T) {
+		config := defaultConfig()
+		config.ProcessorVersion = "v2"
+		config.ConnectionString = "Endpoint=sb://my-namespace.servicebus.windows.net/;SharedAccessKeyName=my-key;SharedAccessKey=my-secret;EntityPath=my-event-hub"
+		config.EventHubName = "my-event-hub"
+		config.SAName = "teststorageaccount"
+		config.SAConnectionString = "DefaultEndpointsProtocol=https;AccountName=teststorageaccount;AccountKey=my-secret;EndpointSuffix=core.windows.net"
+		config.SAContainer = "filebeat-activitylogs-event_hub_00"
+
+		if err := config.Validate(); err != nil {
+			t.Fatalf("unexpected validation error: %v", err)
+		}
+
+		assert.NotNil(t, config.ConnectionStringProperties.EntityPath)
+		assert.Equal(t, config.EventHubName, *config.ConnectionStringProperties.EntityPath)
+	})
+
+	t.Run("Connection string does not contain entity path", func(t *testing.T) {
+		config := defaultConfig()
+		config.ProcessorVersion = "v2"
+		config.ConnectionString = "Endpoint=sb://my-namespace.servicebus.windows.net/;SharedAccessKeyName=my-key;SharedAccessKey=my-secret;"
+		config.EventHubName = "my-event-hub"
+		config.SAName = "teststorageaccount"
+		config.SAConnectionString = "DefaultEndpointsProtocol=https;AccountName=teststorageaccount;AccountKey=my-secret;EndpointSuffix=core.windows.net"
+		config.SAContainer = "filebeat-activitylogs-event_hub_00"
+
+		if err := config.Validate(); err != nil {
+			t.Fatalf("unexpected validation error: %v", err)
+		}
+
+		assert.Nil(t, config.ConnectionStringProperties.EntityPath)
+	})
+
+	t.Run("Connection string contains entity path but does not match event hub name", func(t *testing.T) {
+		config := defaultConfig()
+		config.ProcessorVersion = "v2"
+		config.ConnectionString = "Endpoint=sb://my-namespace.servicebus.windows.net/;SharedAccessKeyName=my-key;SharedAccessKey=my-secret;EntityPath=my-event-hub"
+		config.EventHubName = "not-my-event-hub"
+		config.SAName = "teststorageaccount"
+		config.SAConnectionString = "DefaultEndpointsProtocol=https;AccountName=teststorageaccount;AccountKey=my-secret;EndpointSuffix=core.windows.net"
+		config.SAContainer = "filebeat-activitylogs-event_hub_00"
+
+		err := config.Validate()
+		if err == nil {
+			t.Fatalf("expected validation error")
+		}
+
+		assert.NotNil(t, config.ConnectionStringProperties.EntityPath)
```

Comment:
same, please use `require` to prevent a panic after this check

Reply:
Replaced with `require`.